lambda-source/index.js

```diff
@@ -6,54 +6,55 @@ const FORM_HANDLER = process.env.PARDOT_FORM_HANDLER_URL
 const RECAPTCHA_SECRET = process.env.RECAPTCHA_SECRET
 const REDIRECT_URL = process.env.REDIRECT_URL
 
+// Unused headers?
 const headers = {
   'Content-Type': 'application/json',
   'Access-Control-Allow-Origin': '*',
   'Access-Control-Allow-Headers': "Content-Type",
   "Access-Control-Allow-Methods": "OPTIONS,POST"
 }
 
-exports.handler = async (event, context, callback) => {
-  console.log("Full event body", event.body);
+async function post(url, payload, opts) {
+  return axios.post(url, payload, { ...opts })
+}
 
-  let buff = Buffer.from(event.body, 'base64');
-  let text = buff.toString('ascii');
 
-  var keyValuePairs = text.split('&');
-  var json = {};
+async function verifyRecaptcha(captchaResponse) {
+  try {
+    const response = await post(
+      'https://www.google.com/recaptcha/api/siteverify', 
+      querystring.stringify({
+        secret: RECAPTCHA_SECRET,
+        response: captchaResponse
+      }),
+      {
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded"
+        }
+      }
+    )
+    console.log("Captcha Result: ", response);
+    const { data } = response;
 
-  console.log("Parsing form");
-  for (var i = 0, len = keyValuePairs.length, tmp, key, value; i < len; i++) {
-    tmp = keyValuePairs[i].split('=');
-    key = decodeURIComponent(tmp[0]);
-    value = decodeURIComponent(tmp[1]);
-    if (key.search(/\[\]$/) != -1) {
-      tmp = key.replace(/\[\]$/, '');
-      json[tmp] = json[tmp] || [];
-      json[tmp].push(value);
-    } else {
-      json[key] = value;
+    if (!data || !data.success) {
+      console.error("Invalid Recpature");
```

Comment:
nit: typo in Recaptcha

Reply:
You mean we aren't using `Goggle Recpature` ?